torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -96,13 +97,16 @@ size_t collectMaxVectorizeSizeWithContigMerge(
 
   // Assume no halo-related expression appears in the fusion. No
   // broadcast is merged, so indexability can be assumed to be true.
+  // This is expensive, as ContigIDs builds other things like CAMap,
+  // HaloInfo, and ConcreteBroadcast info. We should explicitly build and reuse
+  // these as they're compile time information.
```

Comment:
What do you think the signature should be? I don't really want to put a "build" call into these classes that are just building const returnable information.

Reply:
I was basically thinking moving below here:
(copied from `ContigIDs::ContigIDs()`)
```
ca_map_ = std::make_shared<ComputeAtMap>(ids[0]->fusion());
    halo_info_ = std::make_shared<HaloInfo>(ids[0]->fusion(), ca_map_);
    concrete_info_ =
        std::make_shared<ConcretizedBroadcastDomains>(ids[0]->fusion());

    consistent_transform_info_ = std::make_unique<const OrderedIdInformation>(
        ids, root_domain, concrete_info_);
```

And then we could use the other constructor of `ContigIDs`.

This isn't something I'm strongly pushing at all, but I'm just feeling the ContigIDs constructors would need some refactoring as there are also calls like `ContigIDs({}, {}, {}, {})`, which doesn't look pretty.

Reply:
Yeah, some cleanup would help, but I was struggling with exact changes to make. I made an empty buildable ContigID constructor too and don't remember why but ended up not liking it either.

Reply:
I'm still not following where you would move the building of ca map, halo map, and concrete bcast map.

Reply:
Do you just want me to force building these by the user of ContigIDs?

Reply:
Yeah, I think that's my preference as it'd make explicit that these expensive operations are done, though it would be tedious especially if ContigIDs would be used more often. I'd be fine with either way.